test/unit/query-generator/create-table-query.test.ts

```diff
@@ -0,0 +1,401 @@
+import { DataTypes } from '@sequelize/core';
+import { expectsql, sequelize } from '../../support';
+
+// TODO: overhaul this test suite when migrating createTableQuery to TypeScript
+/**
+ * Make methods stricter, throw when using invalid options
+ * Accept and test for TableNameOrModel
+ * Check if createTableQuery is typed correctly
+ * Check if all tests make sense, the current tests are just copied from dialect specific tests and other expectations are added
+ * Give tests unique names
+ * Make sure that all resulting queries are valid by adding integration tests for QueryInterface.createTable
+ * Make use of the default expectation and tick replacements instead of setting an expectation for each dialect individually
+ */
+
+describe('QueryGenerator#createTableQuery', () => {
+  const queryGenerator = sequelize.getQueryInterface().queryGenerator;
+
+  it('createTableQuery', () => {
+    expectsql(queryGenerator.createTableQuery('myTable', { int: 'INTEGER' }, {}), {
```

Comment:
Could we change the test names to include the data types? Maybe something like "create table query - integer", doesn't necessarily have to be that though

Reply:
I'll work on some more better naming